scripts/py_matter_idl/matter/idl/generators/cpp/sdk/sdk_generator.py

```diff
@@ -85,6 +86,73 @@ def response_struct(s: Struct) -> bool:
     return s.code is not None
 
 
+@dataclass
+class EnumEntry:
+    namespace: str
+    enum: Enum
+
+
+def extract_shared_enums(idl: Idl) -> list[EnumEntry]:
+    """
+    Retunrs global/shared enumerations, distinguished by namespace
+    """
+    result = []
+    for g in idl.global_enums:
+        result.append(EnumEntry(namespace="Globals", enum=g))
+
+    # shared enums are unique by name
```

Comment:
This is just not true in general sadly.  I mean, it happens to be true for us, but nothing prevents anyone from writing XML where this will not be true (and of course then our codegen will be quite broken).

Reply:
Expanded the comment to explain - we use `detail` as a shared namespace, so if we ever name clash things would break anyway. Not the fault of this code, it just applies what already exists.